### PR TITLE
Add `diff_schema.sh` / `diff_state.sh` tools

### LINKED ISSUE #72: Tools to help build API Migration Guide
We can start with something like:

`git diff v0.5.3 contracts/*/msg.rs` or maybe schema diffs

some tool to get the raw info of what changed, then we can hand-make before/after examples that highlight the changes

### PR DESCRIPTION
**Pull request by maurolacy**
Closes #72.

This is not the best script I ever wrote, but it's a start.

Feel free to suggest improvements / extensions. ~~Will take a look at `jsondiff` and how to use it next.~~ (added a `-j` option for using `jsondiff`)

**Update**: Added a tentative `diff_state.sh` tool.

### CODE REVIEW

File: scripts/diff_state.sh [hunk 1]
```diff
@@ -0,0 +1,78 @@
...
+  mv "$S" "$S-$RIGHT_TAG"
+done
+
+# Compare it
+for SL in */*/"state-$LEFT_TAG"
+do
+  PARENT=$(dirname "$SL")
+  echo "$PARENT":
+  SR="$PARENT/state-$RIGHT_TAG"
+  diff -u <(sed '/^\s*#\[cfg(test)\]/,$d' "$SL"/state.rs) <(sed '/^\s*#\[cfg(test)\]/,$d' "$SR"/state.rs) | { grep -v '^[+-]\s*\/[\/\*]' || true; }
```
Comment (uint) [inline scripts/diff_state.sh:71]: I'm worried about this regexp. I tried this out in https://sed.js.org/ with this input:

```rust
mod query_responses;
#[cfg(test)]
mod state_test_utils;

struct InstantiateMsg {
    code_id: u64,
}
```

I know we don't currently have that in our `state.rs` files, but I can imagine us doing something of the sort in the future, the CI diff breaking and devs being very confused.

**Review by uint — COMMENTED** [review]
Looks like an improvement!

File: scripts/diff_state.sh (hunk 1 above)
Comment (maurolacy) [reply scripts/diff_state.sh:71]: Hmm, an improvement would require multi-line matching, using `ssed` or so. Not simple / trivial to do, and still fragile.

I'm OK with removing that filter.

Comment (maurolacy) [reply scripts/diff_state.sh:71]: Here you go: `sed -E '{N; /^\s*#\[cfg\(test\)\]\s+mod\s+[a-z_]+\s*\{/,$d}'`

**Review by hashedone — APPROVED** [review]
LGTM

I didn't go in details as it is just a tool and wont break too much, looks valid as much as I know bash. I suggest just merging - if anything would need fix, it would be fixed online I think.

**Pull request merged**